arduino-ide-extension/src/browser/contributions/upload-sketch.ts

```diff
@@ -258,7 +163,7 @@ export class UploadSketch extends CoreServiceContribution {
     if (!CurrentSketch.isValid(sketch)) {
       return undefined;
     }
-    const userFields = this.userFields();
+    const userFields = this.userFields.getUserFields();
```

Comment:
Why not pass the `fqbn` from here?

Reply:
I don't see any reason since I can easily get them from the `boardsServiceProvider`. Passing it here would just add a parameter, with no benefit.

Reply:
Exactly this; why do you want to calculate it twice? `verify`, and `upload` will require the `fqbn` anyway.

Reply:
IMO how the user fields are stored is not related to the upload. I may want to open that dialog even if I'm not uploading.